mbed/mbed.py

```diff
@@ -607,8 +607,24 @@ def cleanup():
         for branch in branches: # delete all local branches so the new repo clone is not poluted
             pquery([git_cmd, 'branch', '-D', branch])
 
-    def clone(url, name=None, depth=None, protocol=None):
-        popen([git_cmd, 'clone', formaturl(url, protocol), name] + (['--depth', depth] if depth else []) + (['-v'] if very_verbose else ([] if verbose else ['-q'])))
+    def clone(url, path, rev=None, depth=None, protocol=None, name=None):
+        result = pquery([git_cmd, "ls-remote", url, (rev if rev else "HEAD")])
+        
+        if result and rev:
+            repo_name = url.split('/')[-1]
```

Comment:
This breaks the behavior of clone() regarding param path. As an example if path="/tmp/sometempdir", the first part of this "if" will ignore the path value and create it's own dir based on the url. The expected behavior (as show in the else) is that path is passed to git and therefore the cloned repository will be stored in path, not in a folder generated from the url name.
E.g. imagine that there's no correlation between the library folder name and the repo url, e.g. https://github.com/ARMmbed/spiflash-stm32f429-driver is referenced from spiflash-driver.lib.

Reply:
Ah, that's a good point. It didn't occur to me that the paths could also point to local repos.